src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -311,8 +369,8 @@ public SCMProbeStat stat(@NonNull String path) throws IOException {
                                 continue;
                             }
                         }
-                        SCMHead head = new SCMHead(branchName);
-                        SCMRevision hash = new SCMRevisionImpl(head, b.getSHA1String());
+                        SCMHead head = new SCMHead(name);
```

Comment:
If it is a Tag then should not be returning the `SCMHead` object rather an `SCMHead` that implements the `TagSCMHead` mixin interface.